src/main/scala/collections/dataAnalysis/Reporter.scala

```diff
@@ -0,0 +1,57 @@
+package collections.dataAnalysis
+
+object Reporter {
+
+  def primaryToOtherTimeCorrelation(people: Stream[Person]): String = {
+    val (totalPrimary, totalOther) = people.view.foldLeft((0.0, 0.0))((acc, person) =>
+      (acc._1 + person.timePrimary, acc._2 + person.timeWorking + person.timeOther)
+    )
+    val primPercent = BigDecimal(100 * totalPrimary/ (totalPrimary + totalOther)).setScale(2, BigDecimal.RoundingMode.HALF_UP).toDouble
+    s"$primPercent% : ${100 - primPercent}%"
+  }
+
+  def womenAndMenWorkingTimeComparison(people: Stream[Person]): String = {
+    val (avgFemaleWT, _, avgMaleWT, _) = people.view.foldLeft((0.0, 1.0, 0.0, 1.0))((acc, person) =>
+      person.gender match {
```

Comment:
also consider reduce here

Reply:
Changed in following way:
1) partition for initial stream to split it to male and female streams
2) calculating sum of female stream and dividing it on stream length (sum = reduce(_ + _))
3) same for male stream

Not sure if this is good implementation, cause it looks like we are parsing initial stream 3 time instead of 1 as we had before